### PR TITLE
add: transition functionality from Who is Ill page

### PR DESCRIPTION
**Pull request by akaJSaunders**
### JIRA link

https://companieshouse.atlassian.net/browse/LFA-1443

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/IllPersonController.ts [hunk 1]
```diff
@@ -61,11 +62,21 @@ export class IllPersonController extends BaseController<FormBody> {
...
         if (appeal.reasons?.illness != null) {
+
             appeal.reasons.illness.illPerson = value.illPerson;
+
             if (value.illPerson !== IllPerson.someoneElse) {
+
                 appeal.reasons.illness.otherPerson = undefined;
+
+            } else {
+
```
Comment (rhysbarrett) [inline src/controllers/IllPersonController.ts:75]: I'm curious about all of the additional whitespace, particularly in this `else` block. Is there a style guide I need to take a look at?

Comment (akaJSaunders) [reply src/controllers/IllPersonController.ts:75]: I only did that because when debugging I was finding it hard to read. I think its subjective.

Comment (rhysbarrett) [reply src/controllers/IllPersonController.ts:75]: Fair enough! Just checking :)

Comment (akaJSaunders) [reply src/controllers/IllPersonController.ts:75]: Im not very consistent, but since we had nested if and else, I thought it might be better

**Pull request merged**